### PR TITLE
Update README and installation docs.

### PR DESCRIPTION
**Pull request by aragilar**
This expands and updates the installation docs, and simplifies the installation suggestions in the README. It also removes references to outdated information (Python XY seems dead, we no longer appear to distribute Windows installers).

Opinions?

### CODE REVIEW

**Comment by aragilar** [comment]
Also, it appears we never actually test that the docs build, we should do that...

File: README.rst [hunk 1]
```diff
@@ -13,43 +14,23 @@ Websites
...
+------------
 
-Prerequisites
--------------
+Pre-build `h5py` can either be installed via your Python Distribution (e.g.
+`Continuum Anaconda`_, `Enthought Canopy`_) or from `PyPI`_ via `pip`_.
 
-You need, at a minimum:
+More detailed installation instructions, including how to install `h5py` with
+MPI support, can be found at: http://docs.h5py.org/en/latest/build.html.
```
Comment (tacaswell) [inline README.rst]: I would use internal links here 

```
:ref:`build`
```
here

and 

```
.. _build:
```

over the section you want to link ot.

File: docs/build.rst [hunk 2]
```diff
@@ -3,45 +3,90 @@
...
+Wheels
+......
+If you have an existing Python installation (e.g. a python.org download,
+or one that comes with your OS), then on Windows, MacOS/OSX, and
+Linux on Intel computers, pre-built h5py wheels can be installed via pip from
+PyPI::
 
     $ pip install h5py
 
+Additionally, for Windows users, `Chris Gohlke provides third-party wheels
```
Comment (tacaswell) [inline docs/build.rst]: Cristoph builds the windows wheels we have on pypi for us.  I wonder if those are complied against mkl.

File: README.rst (hunk 1 above)
Comment (aragilar) [inline README.rst]: Can internal links be used when the documents are used in different places (README isn't used by sphinx, only github/PyPI)?

File: docs/build.rst (hunk 2 above)
Comment (aragilar) [inline docs/build.rst]: As far as I know, HDF5 doesn't care about BLAS, so I suspect it's going to come down to how much numpy ABI is affected by using MKL (numpy on PyPI does not use MKL afaik, for the same reason there's no windows scipy wheel).

File: README.rst (hunk 1 above)
Comment (tacaswell) [inline README.rst]: oh, never mind then.

**Pull request merged**